src/util/name_table.cpp

```diff
@@ -0,0 +1,52 @@
+#include "util/name_table.hpp"
+#include "util/simple_logger.hpp"
+
+#include <algorithm>
+#include <limits>
+#include <fstream>
+
+#include <boost/filesystem/fstream.hpp>
+
+namespace osrm
+{
+namespace util
+{
+
+NameTable::NameTable(const std::string &filename)
+{
+    boost::filesystem::ifstream name_stream(filename, std::ios::binary);
+
+    name_stream >> m_name_table;
+
+    unsigned number_of_chars = 0;
+    name_stream.read(reinterpret_cast<char *>(&number_of_chars), sizeof(number_of_chars));
+    BOOST_ASSERT_MSG(0 != number_of_chars, "name file broken");
+    m_names_char_list.resize(number_of_chars + 1); //+1 gives sentinel element
```

Comment:
I honestly didn't think about it. I just copied it from the datafacade. The problem here, I require the names which have been serialized to disk and cannot be accessed later without reeding them back in. I cannot think of a reasonable way to have this in here, but @TheMarex reasonably said that some of these tests should be performed in pre-processing, if possible